packages/headless/src/controllers/triggers/headless-notify-trigger.ts

```diff
@@ -0,0 +1,76 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {ConfigurationSection, TriggerSection} from '../../state/state-sections';
+import {configuration, triggers} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {logNotifyTrigger} from '../../features/triggers/trigger-analytics-actions';
+
+/**
+ * The `NotifyTrigger` controller handles Notify triggers.
+ */
+export interface NotifyTrigger extends Controller {
+  /**
+   * the state of the `NotifyTrigger` controller.
+   */
+  state: NotifyTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `NotifyTrigger` controller.
+ */
+export interface NotifyTriggerState {
+  /**
+   * The notification to present to the user after receiving a notification trigger.
+   */
+  notify: string;
```

Comment:
Instead of a verb for the attribute name here, I would prefer a noun (e.g, `notification` or `message`).